Gem/Code/Source/Components/NetworkWeaponsComponent.cpp

```diff
@@ -375,11 +378,49 @@ namespace MultiplayerSample
+                if ((fireBoneTransform.GetTranslation() - weaponInput->m_shotStartPosition).GetLength() > sv_WeaponsStartPositionClampRange)
+                {
+                    weaponInput->m_shotStartPosition = fireBoneTransform.GetTranslation();
                     AZLOG_WARN("Shot origin was outside of clamp range, resetting to bone position");
                 }
+
+                // Setup a default aim target
+                const WeaponParams weaponParams = GetWeaponParams(weaponIndexInt);
+                AZ::Vector3 aimTarget = worldTm.GetTranslation() + cameraOffset + aimRotation.TransformVector(fwd * weaponParams.m_weaponMaxAimDistance);
+
+                // Cast the ray in the physics system from the center of the camera forward
+                if (auto* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get())
+                {
+                    if (AzPhysics::SceneHandle sceneHandle = sceneInterface->GetSceneHandle(AzPhysics::DefaultPhysicsSceneName);
+                        sceneHandle != AzPhysics::InvalidSceneHandle)
+                    {
+                        AzPhysics::RayCastRequest physicsRayRequest;
+                        physicsRayRequest.m_start = worldTm.GetTranslation() + cameraOffset;
+                        physicsRayRequest.m_direction = aimRotation.TransformVector(fwd);
+                        physicsRayRequest.m_distance = weaponParams.m_weaponMaxAimDistance;
+                        physicsRayRequest.m_queryType = AzPhysics::SceneQuery::QueryType::StaticAndDynamic;
+                        physicsRayRequest.m_reportMultipleHits = true;
+
+                        AzPhysics::SceneQueryHits result = sceneInterface->QueryScene(sceneHandle, &physicsRayRequest);
+                        if (result)
+                        {
+                            for (AzPhysics::SceneQueryHit hit : result.m_hits)
+                            {
+                                // Set target to first found intersect within dot tolerance, if any
+                                AZ::Vector3 targetDirection = hit.m_position - weaponInput->m_shotStartPosition;
+                                AZ::Vector3 aimDirection = physicsRayRequest.m_direction;
+                                targetDirection.Normalize();
+                                aimDirection.Normalize();
+                                if (targetDirection.Dot(aimDirection) > sv_WeaponsDotClamp)
+                                {
+                                    aimTarget = hit.m_position;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }             
+
                 FireParams fireParams{ weaponInput->m_shotStartPosition, aimTarget, Multiplayer::InvalidNetEntityId };
                 TryStartFire(aznumeric_cast<WeaponIndex>(weaponIndexInt), fireParams);
```

Comment:
we should probably not call TryStartFire if the aimTarget is outside the `sv_WeaponsDotClamp` range